lib/scala/logging-config/src/main/java/org/enso/logger/config/LoggerSetup.java

```diff
@@ -0,0 +1,31 @@
+package org.enso.logger.config;
+
+import java.nio.file.Path;
+import org.slf4j.event.Level;
+
+/** Base classes to be implemented by the underlying loggging implementation. */
+public abstract class LoggerSetup {
```

Comment:
`LoggerSetup` seems like a fine abstraction. I am assuming this `lib/scala/logging-config` doesn't depend on anything else than JDK and `slf4j`)...

All this class needs is `static LoggerSetup get()` method that will use `ServiceLoader` to lookup the `LogbackSetup` implementation once on runtime classpath.

Reply:
Yeah, came to the same conclusion after our discussion. I didn't use `ServiceLoader` but some very basic custom reflection code instead, seemed trivial enough to avoid another dependency. Should be fairly easy to switch now to `ServiceLoader` if you feel like there is any benefit of that.

Reply:
> Yeah, came to the same conclusion after our discussion. 

OK, thank you.

> I didn't use `ServiceLoader` but some very basic custom reflection code instead... easy to switch now to 
> `ServiceLoader` if you feel like there is any benefit of that.

According to [Injectable singletons manifesto](http://wiki.apidesign.org/wiki/Injectable_Singleton) there are benefits over plain [reflection](https://github.com/enso-org/enso/pull/7559/commits/76c0efa3ca30ba844a3abe8d05ccfff4ae3a602b#diff-7f876974b7728ca4264314a6e7fac29592eecaa8f266296d5689ab8e25e4eae4R17) including _convention over configuration_, _testability_, reconfiguration by _assembling classpath_, etc. Moreover `ServiceLoader` support has been included into [module-info standard](https://docs.oracle.com/javase%2F9%2Fdocs%2Fapi%2F%2F/java/lang/module/ModuleDescriptor.Builder.html#provides-java.lang.String-java.util.List-). When/if/once we switch to _module path_ instead of _class path_, using `ServiceLoader` will be easier than your home made solution based on properties:
```java
Config c = ConfigFactory.load("enso-logging.conf");
String clazzName = c.getString(implClassKey);
Class<?> clazz = Class.forName(clazzName);
```
As far as I can tell your solution will probably reach the same goals as [Injectable singletons manifesto](http://wiki.apidesign.org/wiki/Injectable_Singleton) stated. The question to ask however is: Why invent our own solution when there is a standard one defined in the JDK?

_Btw: Wasn't the whole purpose of this PR to avoid home made solutions, when there is a standard one?_

Reply:
OK. Done